point_pattern.py

```diff
@@ -93,12 +110,21 @@ def mean_center(points):
     y : float
         Mean y coordinate
     """
-    x = None
-    y = None
+    x = 0
+    y = 0
+    xx = 0
+    yy = 0
+
+    for i in points:
+        xx += i[0]
+        yy += i[1]
+        x = xx / len(points[0])
+        y = yy / len(points[1])
```

Comment:
Watch where you are computing `x` and `y`.  Lines 121 and 122 are executed `i` times right now, but I suspect that you want to sum all of the x coordinates and then divide by `len(points)`.  (Not `len(points[0])`, because a single point will always have a length of 2 - the x and the y coordinate).

As below, feel free to update and resubmit.